zerver/views/messages.py

```diff
@@ -308,7 +306,7 @@ def highlight_string(string, locs):
         result += highlight_stop
         pos = offset + length
     result += string[pos:]
-    return result.decode('utf-8')
+    return result
```

Comment:
I am not convinced that this new logic is correct.  I think it's the case that the locations we have are numbers of bytes in the binary string, not numbers of unicode characters.  So I think we want to `force_bytes()` at the top and `force_text` at the bottom of this function, not remove the encode/decode.

You can check how these offsets are constructed here: https://github.com/zulip/tsearch_extras